v0/destinations/mailchimp/transform.js

```diff
@@ -115,141 +161,169 @@ async function responseBuilderSimple(payload, message, mailChimpConfig) {
-      if (field === "subscriptionStatus") {
-        rawPayload.status = message.integrations.MailChimp[field];
-      } else {
-        rawPayload[field] = message.integrations.MailChimp[field];
-      }
-    });
-    Object.keys(traits).forEach(trait => {
-      if (trait === "email") {
-        rawPayload.email_address = traits[trait];
-      } else if (mailChimpConfig.enableMergeFields) {
-        const tag = filterTagValue(trait);
-        rawPayload.merge_fields[tag] = traits[trait];
-      }
-    });
-    return rawPayload;
-  }
+};
 
+const identifyResponseBuilder = async (message, { Config }) => {
+  const mappedToDestination = get(message, MappedToDestinationKey);
+  const { apiKey, datacenterId, enableMergeFields } = Config;
   const email = getFieldValueFromMessage(message, "email");
-  if (email) {
-    const rawPayload = {};
-    rawPayload.merge_fields = {};
-
-    Object.keys(traits).forEach(trait => {
-      if (trait === "email") {
-        rawPayload.email_address = traits[trait];
-      } else {
-        const tag = filterTagValue(trait);
-        rawPayload.merge_fields[tag] = traits[trait];
-      }
-    });
-    // given email does not exist in the list
-    if (!emailExists) {
-      const isDoubleOptin = await checkIfDoubleOptIn(mailChimpConfig);
-      rawPayload.status = isDoubleOptin
-        ? subscriptionStatus.pending
-        : subscriptionStatus.subscribed;
-    }
-    return rawPayload;
-  }
-  return null;
-}
 
-async function getTransformedJSON(message, mailChimpConfig) {
-  const mappedToDestination = get(message, MappedToDestinationKey);
-  if (mappedToDestination) {
-    addExternalIdToTraits(message);
-    return getFieldValueFromMessage(message, "traits");
+  if (!email) {
+    throw new CustomError("email is required for identify", 400);
   }
+  const primaryPayload = {
+    email_address: email
+  };
 
-  const traits = getFieldValueFromMessage(message, "traits");
+  const audienceId = get(message, "context.MailChimp")
+    ? get(message, "context.MailChimp.listId")
+      ? message.context.MailChimp.listId
+      : Config.audienceId
+    : Config.audienceId;
 
-  const updateSubscription = get(message, "integrations.MailChimp")
-    ? message.integrations.MailChimp
-    : undefined;
+  const emailExists = await checkIfMailExists(
+    apiKey,
+    datacenterId,
+    audienceId,
+    email
+  );
 
-  const email = getFieldValueFromMessage(message, "email");
-  if (!email) {
-    throw new CustomError("email is required for identify", 400);
+  if (mappedToDestination) {
+    // build response with existing traits only
+    addExternalIdToTraits(message);
+    return responseBuilderSimple(
+      getFieldValueFromMessage(message, "traits"),
+      message,
+      Config,
+      audienceId,
+      emailExists
+    );
   }
 
-  const emailExists = await checkIfMailExists(mailChimpConfig, email);
+  const updateSubscription = getIntegrationsObj(message, "mailchimp");
```

Comment:
Please share the expected structure of integrations object here for better understanding

Reply:
done